include/odp/api/spec/dma_types.h

```diff
@@ -397,6 +411,35 @@ typedef struct odp_dma_seg_t {
 		 *  with other data formats. */
 		odp_packet_t packet;
 
+		/** Packet index configuration */
+		struct {
+			/** Packet index
+			 *
+			 *  When odp_dma_transfer_param_t::seg_alloc is used, implementation
+			 *  allocates all required destination packets and this index field can be
+			 *  used to define which packet of the to-be-allocated array of packets is
+			 *  used as the destination for this particular destination segment
+			 *  descriptor. Same index can be reused across subsequent segments if
+			 *  application wishes to transfer to different offsets of the same
+			 *  destination packet. This means that if application for example uses
+			 *  segment index 0 in each destination segment descriptor (with different
+			 *  'offset' values), implementation would allocate only a single packet
+			 *  with data being transferred to different offsets of that single
+			 *  destination packet. Indexes should appear in ascending order.
```

Comment:
To avoid any confusion, perhaps the last sentence could be amended like this:
```
Indexes should appear in ascending order, starting from index 0.
```